docs/blog/2022-03-10-zephyr-3-0.md

```diff
@@ -0,0 +1,220 @@
+
+Existing user config repositories using Github Actions to build will pull down Zephyr 3.0 automatically,
+and should work, fine as is. However, to upgrade to the newer Docker image, you should:
+
+- Open `.github/workflows/build.yml` in your editor/IDE
+- Change `zmkfirmware/zmk-build-arm:2.5` to `zmkfirmware/zmk-build-arm:stable` wherever it is found
+
+:::note
+
+If you created your user config repository a while ago, you may find that your `build.yml` file instead references
+a `zephyr-west-action-arm` custom GitHub Action instead. In this case, the upgrade is not as direct. We suggest that
+instead you [re-create your config repository](/docs/user-setup) to get an updated setup using the new automation
+approach.
+
+:::
+
+### VS Code & Docker (Dev Container)
+
+If you build locally using VS Code & Docker then:
+
+- pull the latest ZMK `main` with `git pull` for your ZMK checkout
+- reload the project
+- if you are prompted to rebuild the remote container, click `Rebuild`
+- otherwise, press `F1` and run `Remote Containers: Rebuild Container`
+- Once the container has rebuilt and reloaded, run `west update` to pull the updated Zephyr version and its dependencies.
+
+Once the container has rebuilt, VS Code will be running the 3.0 Docker image.
+
+### Local Host Development
+
+The following steps will get you building ZMK locally against Zephyr 3.0:
+
+- Run the updated [toolchain installation](/docs/development/setup#toolchain-installation) steps, and once completed, remove the previously installed SDK version (optional, existing SDK should still work)
+- pull the latest ZMK `main` with `git pull` for your ZMK checkout
+- run `west update` to pull the updated Zephyr version and its dependencies
+
+From there, you should be ready to build as normal!
+
+## Board/Shield Changes
+
+### RGB Underglow
+
+Zephyr's WS2812 `led_strip` driver added a new required property. When adding [underglow](/docs/features/underglow#adding-rgb-underglow-to-a-board) to a board, you now must also add a `color-mapping` property, like:
+
+```
+  led_strip: ws2812@0 {
+    compatible = "worldsemi,ws2812-spi";
+    label = "WS2812";
+
+    /* SPI */
+    reg = <0>; /* ignored, but necessary for SPI bindings */
+    spi-max-frequency = <4000000>;
+
+    /* WS2812 */
+    chain-length = <10>; /* number of LEDs */
+    spi-one-frame = <0x70>;
+    spi-zero-frame = <0x40>;
+    color-mapping = <LED_COLOR_ID_GREEN
+                          LED_COLOR_ID_RED
+                          LED_COLOR_ID_BLUE>;
+  };
+```
+
+:::note
+
+Standard WS2812 LEDs use a wire protocol where the bits for the colors green, red, and blue values are sent in that order.
+If your board/shield uses LEDs that require the data sent in a different order, the `color-mapping` property ordering should be changed to match.
+
+:::
+
+All existing boards included in ZMK `main` have already been updated, but for existing or new PRs, or out of tree boards, the new property will need to be added.
```

Comment:
In addition to the instructions and snippet in this section, I think a link to a commit/diff like https://github.com/zmkfirmware/zmk/commit/dccfa270b6aa3b4fb11d5341305d2d0a98d8c906 could help folk to further understand how to make these changes:

```suggestion
All existing boards included in ZMK `main` [have already been updated](https://github.com/zmkfirmware/zmk/commit/dccfa270b6aa3b4fb11d5341305d2d0a98d8c906), but for existing or new PRs, or out of tree boards, the new property will need to be added.
```

Additionally, on Discord @caksoylar mentioned:

> From what I can tell it also applies to all the other changes (display selection, logging, clock configuration, uf2) as well; does it make sense to make that statement instead?

I think there is an appropriate commit that could be linked to for each of those sections too (e.g.  https://github.com/zmkfirmware/zmk/commit/0d11c31b9aeab5b5caf5d961385e166308a45584, https://github.com/zmkfirmware/zmk/commit/1e414be5859c320b79daad190d56e1188aac6a21). I've found that seeing the changes in context like this can really provide useful clues when following the instructions doesn't work out.

Reply:
I'm trying to imagine how to do this... The commits will get new IDs once this PR is merged into ZMK proper. I can link to the commits in *my* repo, but if/when I rebase they change. UGH.

Reply:
Ah. For what it’s worth, I think your instructions are clear enough and the additional links are just a nice extra. If it's not possible to add links at this stage I don't think it's the end of the world. After all, the user is able to browse the changes made in this PR if they do get stuck, which is linked in the opening paragraph of the blog post. 

That said, for added discoverability, it could be helpful to pop a link here in the opening sentence of this section:

> The following changes [have already been completed](https://github.com/zmkfirmware/zmk/pull/1143/commits) for all boards/shields in ZMK main branch. For existing or new PRs, or out of tree boards, the following changes are necessary to properly work with the latest changes.

or

> The following changes have already been completed ([in PR #1143](https://github.com/zmkfirmware/zmk/pull/1143/commits)) for all boards/shields in ZMK main branch. For existing or new PRs, or out of tree boards, the following changes are necessary to properly work with the latest changes.